test/compatibility/test_fast_truncate_info.py

```diff
@@ -0,0 +1,170 @@
+
+    def generate_older_database_files(self):
+        import wiredtiger
+
+        conn = wiredtiger.wiredtiger_open('.', 'create,')
+        session = conn.open_session()
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+        session.create(self.uri, self.config + ',key_format=i,value_format=S')
+
+        c = session.open_cursor(self.uri)
+        for i in range(1, self.nentries):
+            c[i] = 'i' + str(i)
+        c.close()
+
+        session.checkpoint()
+
+        session.close()
+        conn.close
+
+    def write_fast_truncate_info(self):
+        import wiredtiger
+        from wiredtiger import stat
+
+        # Open the connection on the database directory containing the database
+        # generated by the older binary.
+        new_conn = wiredtiger.wiredtiger_open('.', 'create=false,statistics=(all)')
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+
+        new_conn.set_timestamp('oldest_timestamp='+ self.timestamp_str(200))
+        new_conn.set_timestamp('stable_timestamp='+ self.timestamp_str(200))
+
+        session = new_conn.open_session()
+        session.begin_transaction()
+        
+        # Truncate a big range of rows.
+        start = session.open_cursor(self.uri, None)
+        start.set_key(10)
+        end = session.open_cursor(self.uri, None)
+        end.set_key(self.nentries)
+        session.truncate(None, start, end, None)
+        start.close()
+        end.close()
+
+        session.commit_transaction('commit_timestamp=' + self.timestamp_str(250))
+
+        # Ensure that fast-truncate happened.
+        stat_cursor = session.open_cursor('statistics:', None, None)
+        fastdelete_pages = stat_cursor[stat.conn.rec_page_delete_fast][2]
+        assert(fastdelete_pages > 0)
+
+        # Move the stable timestamp ahead of the fast-truncate timestamp so that
+        # recovery will not remove this fast-truncate information.
+        new_conn.set_timestamp('stable_timestamp='+ self.timestamp_str(300))
+
+        # Checkpoint, forcing address-deleted cells to be written.
+        session.checkpoint()
+
+        session.close()
+        new_conn.close()
+
+        # Re-open the connection and verify the object.
+        new_conn = wiredtiger.wiredtiger_open('.', 'create=false')
+        session = new_conn.open_session()
+        session.verify(self.uri)
+
+        session.close()
+        new_conn.close()
+
+    def read_fast_truncate_info_from_older(self):
+        import wiredtiger
+
+        # Open the connection on the database directory containing the database
+        # generated by the newer binary which contains fast-truncate information.
+        new_conn = wiredtiger.wiredtiger_open('.', 'create=false')
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+
+        # Open a cursor and update a record to diry the tree.
+        session = new_conn.open_session()
+        c = session.open_cursor(self.uri)
+        c.set_key(5)
```

Comment:
As part of the verification, can you read a single key in the middle of the truncated range? As the range is bigger, this page must be present on a fast truncated page and it shouldn't be visible.